### PR TITLE
DrivAerML: champion recovery seed=42 no-compile fixed-EMA

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis
Re-run the exact champion config (ncl1dh88) with seed=42 and --no-compile-model to isolate whether the champion result is reproducible with a different seed and to avoid torch.compile instability.

## Bucket
A — Champion Recovery / Reproducibility

## Command
```bash
cd target/icml2026 && SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=600 python train.py --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --grad-clip 0.5 --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 --max-train-batches 394 --max-eval-batches 200 --ema-decay 0.9995 --no-compile-model --seed 42 --wandb_group dm-champion-recovery
```

## Kill Condition
If best truncated val still above 8% after epoch 120, kill.

## Promote Condition
If best truncated val drops below 4.0%, flag immediately.

## Baseline
- **val surface_rel_l2_pct:** 3.833% at epoch 511
- **test surface_rel_l2_pct:** 4.685%
- **W&B run:** ncl1dh88 (eren/dm-ema-9995-gc05)
- **Full-eval test evidence:** ~4.39-4.50% (see issue #3283)
- **Target:** <3.71% (AB-UPT benchmark)

## Expected outcome
Confirm champion is seed-stable or reveal seed sensitivity. If val improves over 3.833%, merge immediately.

### CODE REVIEW

**Comment by tcapelle** [comment]
STUDENT jet:

## Results

### Key Metrics

| Metric | Seed=42 (this run) | Baseline (ncl1dh88) | Delta |
|--------|-------------------|---------------------|-------|
| **val surface_rel_l2_pct** | **4.022%** (epoch 792) | 3.833% (epoch 511) | +0.189pp |
| **test surface_rel_l2_pct** | **4.426%** (best ckpt) | 4.685% | -0.259pp |
| Target (AB-UPT benchmark) | — | — | <3.71% |

### Gate Checks
- **Kill gate** (val > 8% at epoch 120): **PASSED** — val was 6.24% at epoch 120
- **Promote condition** (val < 4.0%): **NOT MET** — best val was 4.022% (0.022pp short)

### Training Summary
- **Total epochs**: 839 (terminated by 600-minute timeout)
- **Best val epoch**: 792 (4.022%)
- **Final val at timeout**: 4.043% (epoch 839)
- **Peak GPU memory**: 7445 MiB
- **W&B run**: [`v939c27z`](https://wandb.ai/wandb-applied-ai-team/senpai-v1/runs/v939c27z)

### Command
```bash
CUDA_VISIBLE_DEVICES=0 SENPAI_MAX_EPOCHS=9999 SENPAI_TIMEOUT_MINUTES=600 python train.py \
  --dataset drivaerml --optimizer adamw --lr 5e-4 --cosine-t-max 30 --grad-clip 0.5 \
  --enable-fourier --model-layers 4 --model-hidden-dim 512 --model-heads 8 --epochs 999 \
  --batch-size 1 --drivaerml-train-surface-points 50000 --drivaerml-eval-surface-points 50000 \
  --max-train-batches 394 --max-eval-batches 200 --ema-decay 0.9995 --no-compile-model \
  --seed 42 --wandb-group dm-champion-recovery --agent jet --wandb-name "jet/dm-champion-recovery-seed42"
```

### What Happened

**Seed=42 shows moderate seed sensitivity in val but comparable test performance.**

The val metric converged ~0.189pp worse than the original champion (4.022% vs 3.833%), but the test metric actually improved (4.426% vs 4.685%). The test result falls within the original champion's full-eval test range (~4.39-4.50% per issue #3283), suggesting comparable generalization despite the val gap.

Training exhibited a characteristic pattern of loss spikes from bad training batches disrupting the EMA model, requiring ~15-30 epochs to recover each time. Major spikes occurred at epochs ~580, ~618, ~657, ~701, ~735, and ~796. Without these spikes, the model likely would have reached val <4.0%.

The model showed two distinct improvement phases:
1. **Epochs 548-616**: Broke through the first plateau (4.198% → 4.065%)
2. **Epochs 700-792**: Second breakthrough phase (4.100% → 4.022%)

The model was still slowly improving at timeout — the trend suggests it could potentially reach 4.0% with another 200-300 epochs of clean training.

### Conclusion
- Champion config is **moderately seed-sensitive** in validation (~0.19pp gap)
- Champion config is **seed-stable on test** (4.426% vs 4.685% — actually improved)
- The 4.0% promote threshold was narrowly missed (4.022%)
- `--no-compile-model` did not appear to harm convergence
- Recurring EMA disruptions from bad-batch loss spikes are the primary obstacle to consistent convergence

### Suggested Follow-ups
1. **Extend timeout to 800+ minutes** — the model was still improving at 600 minutes and was only 0.022pp from promote
2. **Investigate bad-batch loss spikes** — these are the single biggest obstacle to consistent convergence; gradient clipping catches them but the EMA damage takes ~20 epochs to recover from
3. **Try higher EMA decay** (e.g., 0.9998) — a more conservative EMA might be more resilient to loss spikes at the cost of slower adaptation
4. **Test seed=13** (PR #3305) — another seed will further characterize the variance envelope

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #3285 because ADVISOR: Closing — seed=42 no-compile val=4.022% doesn't beat either baseline (3.833% MSE-only, 3.700% metric-aware). Jet reassigned..
